static/js/components/QuickSearchBar.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useHistory } from "react-router-dom";
+
+import Link from '@material-ui/core/Link';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import SearchIcon from '@material-ui/icons/Search';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+import { GET } from '../API';
+
+function useDebouncer(value, delay) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(
+    () => {
+      const handler = setTimeout(() => {
+        setDebouncedValue(value);
+      }, delay);
+      return () => {
+        clearTimeout(handler);
+      };
+    },
+    [value, delay]
+  );
+  return debouncedValue;
+}
+
+const QuickSearchBar = () => {
+  const dispatch = useDispatch();
+  const [options, setOptions] = React.useState([]);
+  const [value, setValue] = useState(null);
+  const [inputValue, setInputValue] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const history = useHistory();
+
+  const debouncedInputValue = useDebouncer(inputValue, 250);
+
+
+  useEffect(() => {
+    const get = (val) => dispatch(
+      GET(`/api/sources?sourceID=${val}&pageNumber=1&totalMatches=25`, 'skyportal/FETCH_SOURCES')
+    );
+    (async () => {
+      if (debouncedInputValue === '') {
+        setOptions([]);
+        setOpen(false);
+        return undefined;
+      }
+
+      setLoading(true);
+
+      const response = await get(debouncedInputValue);
+
+      setLoading(false);
+      setOpen(true);
+      const matchingSources = await response.data.sources;
+
+      let newOptions = [];
+
+      if (matchingSources) {
+        const rez = Object.keys(matchingSources).map((key) => matchingSources[key].id);
+        newOptions = [...newOptions, ...rez];
+      }
+      setOptions(newOptions);
+      return undefined;
```

Comment:
Is this to make the linter happy?

Reply:
yes. An an anonymous async function still needs to return something. I added this to make the linter happy.

Reply:
Ok, you may also use single line eslint disables to keep it from complaining.

Reply:
I am ok with complying with this expectation. Leaving in the return statement.